src/Microsoft.Framework.DesignTimeHost/ApplicationContext.cs

```diff
@@ -490,6 +497,34 @@ private bool PerformCompilation()
             return true;
         }
 
+        private bool PerformPluginWork()
+        {
+            if (_pluginRegistration.WasAssigned ||
+                _pluginWorkNeeded.WasAssigned ||
+                _filesChanged.WasAssigned ||
+                _rebuild.WasAssigned ||
+                _restoreComplete.WasAssigned ||
+                _sourceTextChanged.WasAssigned)
```

Comment:
I can expose a `WorkNeeded` property on the `PluginHandler` to short-circuit all of this. I just figured the extra checking here was pretty quick and the `Process` method basically no-ops if there's nothing to do.